### ISSUE TITLE
Add support for using youtube-dl to playback files from various streaming sites

### ISSUE THREAD

**Issue by cnelson** [cnelson/python-airplay#3]
If we have youtube-dl installed, use it to fetch content for us for the sites it supports.

**Comment by cnelson** [comment]
# Implementation Sketch:

I want to be able to do things like:

```
$ airplay https://www.youtube.com/watch?v=lDba6HBbmqE
```

Once #2 is completed, and we have ffmpeg in our pipeline we can use youtube-dl to extract video URLs for the streaming services it supports.  It's very simple:

```
>>> import youtube_dl
>>> y = youtube_dl.YoutubeDL()
>>> y.extract_info('https://www.youtube.com/watch?v=lDba6HBbmqE', download=False)
```

Then we can pass those video formats to ffmpeg for transcode (if needed!) just like any other direct video link.

## TL;DR

![image](https://cloud.githubusercontent.com/assets/604163/13367804/173f109a-dc9a-11e5-95c8-ff68d093d0c6.png)
